tests/QuikGraph.Tests/Algorithms/Contracts/VertexColorizerContract.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System;
+using NUnit.Framework;
+using QuikGraph.Algorithms;
+using QuikGraph.Algorithms.MaximumFlow;
+using QuikGraph.Algorithms.RandomWalks;
+using QuikGraph.Algorithms.Search;
+using QuikGraph.Algorithms.ShortestPath;
+using QuikGraph.Algorithms.TSP;
+using QuikGraph.Tests.Algorithms.MaximumFlow;
+using QuikGraph.Tests.Algorithms.RandomWalks;
+using QuikGraph.Tests.Algorithms.Search;
+using QuikGraph.Tests.Algorithms.ShortestPath;
+using QuikGraph.Tests.Algorithms.TSP;
+
+// ReSharper disable ReturnValueOfPureMethodIsNotUsed
+
+namespace QuikGraph.Tests.Algorithms.Contracts
+{
+    [TestFixtureSource(typeof(Feeder), nameof(Feeder.VertexColorizers))]
+    public class VertexColorizerContract
+    {
+        private readonly Type _classUnderTest;
+
+        public VertexColorizerContract(Type testClassType)
+        {
+            _classUnderTest = testClassType;
+        }
+
+        [SetUp]
+        public void IgnoreIfInterfaceIsNotImplemented()
+        {
+            if (_classUnderTest == typeof(TSP<,,>))
+                Assert.Ignore("The TSP algorithm does not implement colorizer functionality");
+        }
+
+        [Test]
+        public void ExceptionThrown_WhenVertexDoesNotExistInGraph()
+        {
+            var scenario = new ContractScenario
+            {
+                EdgesInGraph = new[] {new Edge<int>(1, 2)},
+                AccessibleVerticesFromRoot = new[] {2},
+                Root = 1,
+                DoComputation = true
+            };
+
+            var algorithm = CreateAlgorithmAndMaybeDoComputation(scenario);
+
+            Assert.Throws<VertexNotFoundException>(() => algorithm.GetVertexColor(3));
+        }
+
+        [Test]
+        public void ExceptionThrown_WhenAlgorithmHasNotYetBeenComputed()
+        {
+            var scenario = new ContractScenario
+            {
+                EdgesInGraph = new[] {new Edge<int>(1, 2)},
+                SingleVerticesInGraph = new int[0],
+                AccessibleVerticesFromRoot = new[] {2},
+                Root = 1,
+                DoComputation = false
+            };
+
+            var algorithm = CreateAlgorithmAndMaybeDoComputation(scenario);
+
+            var expectedExceptionType = GetExpectedExceptionType();
+            Assert.Throws(expectedExceptionType, () => algorithm.GetVertexColor(2));
+
+            Type GetExpectedExceptionType()
```

Comment:
Since exceptions are differents, there is some harmonization to do there!

Reply:
Do you mean that we need to make more changes in this PR or that we should harmonize how the algorithms behave?